redbot/cogs/warnings/warnings.py

```diff
@@ -342,45 +342,57 @@ def __init__(self, bot: Red):
 
     @commands.command()
     @commands.guild_only()
-    async def warnings(
-        self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
-    ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
-
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
-        if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+    @checks.admin()
+    async def warnings(self, ctx: commands.Context, user: Union[discord.Member, int] = None):
+        """List the warnings for the specified user."""
+        msg = ""
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+            if not user_warnings.keys():  # no warnings for the user
+                await ctx.send(_("That user has no warnings!"))
+            else:
+                for key in user_warnings.keys():
+                    mod = ctx.guild.get_member(user_warnings[key]["mod"])
+                    if mod is None:
+                        mod = discord.utils.get(
+                            ctx.guild.get_member(), id=user_warnings[key]["mod"]
```

Comment:
Not really what I meant by that...

My point wasn't to try and make `guild.get_member` work as part of utils.get (this is already properly handled above), but that while this was being changed, we should avoid leaking info from other guilds and not attempt to match outside the current guild.